bukkit/src/main/java/us/myles/ViaVersion/bukkit/handlers/ChannelHandlerContextWrapper.java

```diff
@@ -0,0 +1,253 @@
+        base.fireChannelActive();
+        return this;
+    }
+
+    @Override
+    public ChannelHandlerContext fireChannelInactive() {
+        base.fireChannelInactive();
+        return this;
+    }
+
+    @Override
+    public ChannelHandlerContext fireExceptionCaught(Throwable throwable) {
+        base.fireExceptionCaught(throwable);
+        return this;
+    }
+
+    @Override
+    public ChannelHandlerContext fireUserEventTriggered(Object o) {
+        base.fireUserEventTriggered(o);
+        return this;
+    }
+
+    @Override
+    public ChannelHandlerContext fireChannelRead(Object o) {
+        base.fireChannelRead(o);
+        return this;
+    }
+
+    @Override
+    public ChannelHandlerContext fireChannelReadComplete() {
+        base.fireChannelReadComplete();
+        return this;
+    }
+
+    @Override
+    public ChannelHandlerContext fireChannelWritabilityChanged() {
+        base.fireChannelWritabilityChanged();
+        return this;
+    }
+
+    @Override
+    public ChannelFuture bind(SocketAddress socketAddress) {
+        return base.bind(socketAddress);
+    }
+
+    @Override
+    public ChannelFuture connect(SocketAddress socketAddress) {
+        return base.connect(socketAddress);
+    }
+
+    @Override
+    public ChannelFuture connect(SocketAddress socketAddress, SocketAddress socketAddress1) {
+        return base.connect(socketAddress, socketAddress1);
+    }
+
+    @Override
+    public ChannelFuture disconnect() {
+        return base.disconnect();
+    }
+
+    @Override
+    public ChannelFuture close() {
+        return base.close();
+    }
+
+    @Override
+    public ChannelFuture deregister() {
+        return base.deregister();
+    }
+
+    @Override
+    public ChannelFuture bind(SocketAddress socketAddress, ChannelPromise channelPromise) {
+        return base.bind(socketAddress, channelPromise);
+    }
+
+    @Override
+    public ChannelFuture connect(SocketAddress socketAddress, ChannelPromise channelPromise) {
+        return base.connect(socketAddress, channelPromise);
+    }
+
+    @Override
+    public ChannelFuture connect(SocketAddress socketAddress, SocketAddress socketAddress1, ChannelPromise channelPromise) {
+        return base.connect(socketAddress, socketAddress1, channelPromise);
+    }
+
+    @Override
+    public ChannelFuture disconnect(ChannelPromise channelPromise) {
+        return base.disconnect(channelPromise);
+    }
+
+    @Override
+    public ChannelFuture close(ChannelPromise channelPromise) {
+        return base.close(channelPromise);
+    }
+
+    @Override
+    public ChannelFuture deregister(ChannelPromise channelPromise) {
+        return base.deregister(channelPromise);
+    }
+
+    @Override
+    public ChannelHandlerContext read() {
+        base.read();
+        return this;
+    }
+
+    @Override
+    public ChannelFuture write(Object o) {
+        if (o instanceof ByteBuf && isLatestVersion) {
+            buffer.writeBytes((ByteBuf) o);
```

Comment:
Would this not create malformed packets?

(One glued to another) if it writes multiple?